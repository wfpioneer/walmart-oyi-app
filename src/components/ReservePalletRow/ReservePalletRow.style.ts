import { StyleSheet } from 'react-native';
import COLOR from '../../themes/Color';

export default StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'flex-start',
    backgroundColor: COLOR.WHITE,
    paddingHorizontal: 15,
    paddingVertical: 5,
    marginBottom: 2,
    flexDirection: 'row'
  },
  content: {
    alignItems: 'flex-start',
    justifyContent: 'space-evenly',
    flex: 1
  },
  itemNbr: {
    color: COLOR.GREY_700,
    fontSize: 12,
    lineHeight: 22
  },
  palletCreateTs: {
    color: COLOR.GREY_500,
    fontSize: 12
  },
  moreText: {
    color: COLOR.GREY_500,
    fontSize: 12
  },
  price: {
    color: COLOR.GREY_500,
    lineHeight: 22
  },
  itemContainer: {
    paddingLeft: 15,
    paddingVertical: 15
  }
});
