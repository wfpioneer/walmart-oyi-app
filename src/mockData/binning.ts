export const mockPallets = [{
  id: 123456,
  expirationDate: '10/3/2022',
  firstItem: {
    itemDesc: 'itemDesc',
    price: 123,
    upcNbr: '12343534',
    itemNbr: 351231,
    quantity: 2
  }
}, {
  id: 345345,
  expirationDate: '10/3/2022',
  firstItem: {
    itemDesc: 'itemDesc',
    price: 123,
    upcNbr: '345345',
    itemNbr: 351231,
    quantity: 2
  }
}];
