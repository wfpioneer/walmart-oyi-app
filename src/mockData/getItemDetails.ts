import ItemDetails from '../models/ItemDetails';
/* eslint-disable quote-props */

const floorType = 'Sales Floor';
const salesLastUpdateTs = '2020-07-15T08:02:17-05:00';
const salesDay1 = '2020-07-08';
const salesDay2 = '2020-07-09';
const salesDay3 = '2020-07-10';
const salesDay4 = '2020-07-11';
const salesDay5 = '2020-07-12';
const salesDay6 = '2020-07-13';
const salesDay7 = '2020-07-14';

export default {
  '123': {
    itemName: 'Test Item That is Really, Really Long (and has parenthesis)',
    itemNbr: 1234567890,
    upcNbr: '000055559999',
    status: 'Active',
    categoryNbr: 93,
    categoryDesc: 'Meat PI',
    price: 2000.94,
    basePrice: 1500.30,
    exceptionType: 'nsfl',
    imageUrlKey: undefined,
    imageBlobKey: undefined,
    completed: false,
    onHandsQty: 42,
    pendingOnHandsQty: -999,
    consolidatedOnHandQty: 10,
    claimsOnHandQty: 5,
    backroomQty: 6,
    cloudQty: undefined,
    replenishment: {
      onOrder: 48
    },
    location: {
      floor: [
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 1,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '1',
          locationName: 'A1-1',
          type: floorType,
          typeNbr: 8
        },
        {
          zoneId: 0,
          aisleId: 0,
          sectionId: 2,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '2',
          locationName: 'A1-2',
          type: 'End Cap',
          typeNbr: 12
        },
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 3,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '3',
          locationName: 'A1-3',
          type: 'Pod',
          typeNbr: 13
        },
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 4,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '4',
          locationName: 'A1-4',
          type: 'Display',
          typeNbr: 11
        }
      ],
      reserve: [
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 1,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '1',
          locationName: 'A1-1',
          type: 'Reserve',
          typeNbr: 7
        }
      ],
      count: 5
    },
    sales: {
      lastUpdateTs: salesLastUpdateTs,
      dailyAvgSales: 15,
      daily: [
        {
          day: salesDay1,
          value: 100
        },
        {
          day: salesDay2,
          value: 0
        },
        {
          day: salesDay3,
          value: 10
        },
        {
          day: salesDay4,
          value: 10
        },
        {
          day: salesDay5,
          value: 42
        },
        {
          day: salesDay6,
          value: 5
        },
        {
          day: salesDay7,
          value: 1
        }
      ],
      weeklyAvgSales: 10,
      weekly: [
        {
          week: 51,
          value: 39
        },
        {
          week: 52,
          value: 26
        },
        {
          week: 53,
          value: 45
        },
        {
          week: 54,
          value: 34
        },
        {
          week: 55,
          value: 34
        }
      ]
    }
  } as ItemDetails,
  '456': {
    itemName: 'Small, Store Use Item',
    itemNbr: 987654321,
    upcNbr: '777555333',
    status: 'Active',
    categoryNbr: 99,
    categoryDesc: 'Store Use',
    price: 2000.94,
    basePrice: 1500.30,
    exceptionType: 'nsfl',
    imageUrlKey: undefined,
    imageBlobKey: undefined,
    completed: false,
    onHandsQty: 42,
    pendingOnHandsQty: -999,
    consolidatedOnHandQty: 0,
    claimsOnHandQty: 5,
    backroomQty: 6,
    cloudQty: 42,
    replenishment: {
      onOrder: 48
    },
    location: {
      floor: [
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 1,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '1',
          locationName: 'A1-1',
          type: floorType,
          typeNbr: 8
        },
        {
          zoneId: 0,
          aisleId: 0,
          sectionId: 2,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '2',
          locationName: 'A1-2',
          type: 'End Cap',
          typeNbr: 12
        },
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 3,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '3',
          locationName: 'A1-3',
          type: 'Pod',
          typeNbr: 13
        },
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 4,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '4',
          locationName: 'A1-4',
          type: 'Display',
          typeNbr: 11
        }
      ],
      reserve: [
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 1,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '1',
          locationName: 'A1-1',
          type: 'Reserve',
          typeNbr: 7
        }
      ],
      count: 5
    },
    sales: {
      lastUpdateTs: salesLastUpdateTs,
      dailyAvgSales: 15,
      daily: [
        {
          day: salesDay1,
          value: 10
        },
        {
          day: salesDay2,
          value: 0
        },
        {
          day: salesDay3,
          value: 10
        },
        {
          day: salesDay4,
          value: 10
        },
        {
          day: salesDay5,
          value: 42
        },
        {
          day: salesDay6,
          value: 5
        },
        {
          day: salesDay7,
          value: 1
        }
      ],
      weeklyAvgSales: 10,
      weekly: [
        {
          week: 51,
          value: 0
        },
        {
          week: 1,
          value: 10
        },
        {
          week: 2,
          value: 27
        },
        {
          week: 3,
          value: 10
        }
      ]
    }
  } as ItemDetails,
  '789': {
    itemName: 'Wine Item',
    itemNbr: 987654321,
    upcNbr: '777555333',
    status: 'Active',
    categoryNbr: 19,
    categoryDesc: 'Wine',
    price: 2000.94,
    basePrice: 1500.30,
    exceptionType: 'nsfl',
    imageUrlKey: undefined,
    imageBlobKey: undefined,
    completed: false,
    onHandsQty: 42,
    consolidatedOnHandQty: 0,
    claimsOnHandQty: 0,
    backroomQty: 6,
    pendingOnHandsQty: 35,
    cloudQty: 42,
    replenishment: {
      onOrder: 48
    },
    location: {
      floor: [
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 1,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '1',
          locationName: 'A1-1',
          type: floorType,
          typeNbr: 8
        },
        {
          zoneId: 0,
          aisleId: 0,
          sectionId: 2,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '2',
          locationName: 'A1-2',
          type: 'End Cap',
          typeNbr: 12
        },
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 3,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '3',
          locationName: 'A1-3',
          type: 'Pod',
          typeNbr: 13
        },
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 4,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '4',
          locationName: 'A1-4',
          type: 'Display',
          typeNbr: 11
        }
      ],
      reserve: [
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 1,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '1',
          locationName: 'A1-1',
          type: 'Reserve',
          typeNbr: 7
        }
      ],
      count: 5
    },
    sales: {
      lastUpdateTs: salesLastUpdateTs,
      dailyAvgSales: 15,
      daily: [
        {
          day: salesDay1,
          value: 10
        },
        {
          day: salesDay2,
          value: 0
        },
        {
          day: salesDay3,
          value: 10
        },
        {
          day: salesDay4,
          value: 10
        },
        {
          day: salesDay5,
          value: 42
        },
        {
          day: salesDay6,
          value: 5
        },
        {
          day: salesDay7,
          value: 1
        }
      ],
      weeklyAvgSales: 10,
      weekly: [
        {
          week: 51,
          value: 0
        },
        {
          week: 1,
          value: 10
        },
        {
          week: 2,
          value: 27
        },
        {
          week: 3,
          value: 10
        }
      ]
    }
  } as ItemDetails
};
