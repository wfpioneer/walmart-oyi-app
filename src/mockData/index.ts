
import itemDetails from './getItemDetails';


export const getMockItemDetails = (upc: string) => {
  // @ts-ignore
  return itemDetails[upc] || {};
};
