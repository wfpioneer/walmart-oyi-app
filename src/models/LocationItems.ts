export interface ZoneItem {
  zoneId: number;
  zoneName: string;
  aisleCount: number;
}

export interface AisleItem {
  aisleId: number;
  aisleName: string;
  sectionCount: number;
}

export interface SectionItem {
  sectionId: number;
  sectionName: string;
}

export interface SectionDetailsItem {
  itemNbr: number;
  itemDesc: string;
  price: number;
}

export interface SectionDetailsPallet {
  palletId: number;
  palletCreateTS: string;
  items?: Omit<SectionDetailsItem, 'price'>[];
}

export interface ReserveDetailsPallet{
  id: number;
  palletCreateTS?: string;
  items: SectionDetailsItem[];
  statusCode: number;
}

export interface LocationItem {
  zone: {
    id: number;
    name: string;
  };
  aisle: {
    id: number;
    name: string;
  };
  section: {
    id: number;
    name: string;
  };
  items: { sectionItems: SectionDetailsItem[] };
  pallets: {
    palletData: Omit<SectionDetailsPallet, 'items'>[];
  }
}

export interface PossibleZone {
  name: string;
  description: string;
}

// eslint-disable-next-line no-shadow
export enum CREATE_FLOW {
  CREATE_ZONE = 'CREATE_ZONE',
  CREATE_AISLE = 'CREATE_AISLE',
  CREATE_SECTION = 'CREATE_SECTION',
  NOT_STARTED = 'NOT_STARTED'
}
