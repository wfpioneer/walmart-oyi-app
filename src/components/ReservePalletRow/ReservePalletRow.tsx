import React from 'react';
import { Image, Text, View } from 'react-native';
import { useTypedSelector } from '../../state/reducers/RootReducer';
import styles from './ReservePalletRow.style';
import { strings } from '../../locales';
import { SectionDetailsPallet } from '../../models/LocationItems';

export type ReservePalletRowProps = { reservePallet: SectionDetailsPallet };
const ReservePalletRow = (props: ReservePalletRowProps): JSX.Element => {
  const { reservePallet } = props;
  const userFeatures = useTypedSelector(state => state.User.features);
  return (
    <View style={styles.container}>
      <View style={styles.content}>
        <Text>
          {`${strings('LOCATION.PALLET')} ${reservePallet.palletId}`}
        </Text>
        <Text style={styles.palletCreateTs}>
          {`${strings('LOCATION.CREATED_ON')} ${reservePallet.palletCreateTS}`}
        </Text>
        { reservePallet.items.length > 0 && (
        <View style={styles.itemContainer}>
          <Text style={styles.itemNbr}>
            {`${strings('ITEM.ITEM')} ${reservePallet.items[0]?.itemNbr}`}
          </Text>
          <Text>
            {reservePallet.items[0].itemDesc}
          </Text>
          { reservePallet.items.length > 1
              && (
              <Text style={styles.moreText}>
                {`+${reservePallet.items.length - 1} ${strings('LOCATION.MORE')}`}
              </Text>
              )}
          { userFeatures.includes('location management edit') && (
          <View style={styles.trashIcon}>
            <Image
              source={require('../../assets/images/trash_can.png')}
            />
          </View>
          )}
        </View>
        )}
      </View>
    </View>
  );
};

export default ReservePalletRow;
