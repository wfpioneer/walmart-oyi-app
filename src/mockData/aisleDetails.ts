import { AisleItem } from '../models/LocationItem';

export const mockAisles: AisleItem[] = [
  {
    aisleId: 1,
    aisleName: 'Aisle 1',
    sectionCount: 30
  },
  {
    aisleId: 2,
    aisleName: 'Aisle 2',
    sectionCount: 0
  },
  {
    aisleId: 3,
    aisleName: 'Aisle 3',
    sectionCount: 2
  },
  {
    aisleId: 4,
    aisleName: 'Aisle 4',
    sectionCount: 2
  },
  {
    aisleId: 5,
    aisleName: 'Aisle 5',
    sectionCount: 30
  },
  {
    aisleId: 6,
    aisleName: 'Aisle 6',
    sectionCount: 0
  }
];
