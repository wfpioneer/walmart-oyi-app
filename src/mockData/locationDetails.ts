import { LocationItem } from '../models/LocationItems';

export const mockLocationDetails : LocationItem = {
  zone: {
    id: 1,
    name: 'G - Grocery'
  },
  aisle: {
    id: 1,
    name: 'Aisle 1'
  },
  section: {
    id: 1,
    name: 'Section 1'
  },
  floor: [
    {
      itemNbr: 980078597,
      itemDesc: 'Nature Valley Sweet and Salty Granola Bars',
      price: 2.40
    },
    {
      itemNbr: 980011111,
      itemDesc: 'Nature Valley Honey Bars',
      price: 5.80
    }

  ],
  reserve: [
    {
      palletId: 1,
      palletCreateTS: '2021-1-1',
      items: [
        {
          itemNbr: 1,
          itemDesc: 'Cabbage',
          price: 2.40
        }
      ]
    }
  ]
};

export const mockLocationDetailsEmpty : LocationItem = {
  ...mockLocationDetails,
  floor: [],
  reserve: []
};

export const mockLocationDetailsLargeLocationCount : LocationItem = {
  ...mockLocationDetails,
  floor: Array.from({ length: 100 }, () => mockLocationDetails.floor[0]),
  reserve: Array.from({ length: 100 }, () => mockLocationDetails.reserve[0])
};
