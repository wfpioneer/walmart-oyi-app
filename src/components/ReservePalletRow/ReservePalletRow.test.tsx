import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { Provider } from 'react-redux';
import { mockLocationDetails } from '../../mockData/locationDetails';
import ReservePalletRow from './ReservePalletRow';
import { SectionDetailsItem, SectionDetailsPallet } from '../../models/LocationItems';
import store from '../../state';

const mockReservePallet = mockLocationDetails.pallets.palletData[0];
const mockSection = mockLocationDetails.section;

jest.mock('react-redux', () => {
  const ActualReactRedux = jest.requireActual('react-redux');
  return {
    ...ActualReactRedux,
    useTypedSelector: jest.fn().mockImplementation(() => { })
  };
});
describe('ReservePalletRow Component', () => {
  it('Renders a ReservePallet', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <Provider store={store}>
        <ReservePalletRow sectionId={mockSection.id} reservePallet={mockReservePallet} />
      </Provider>
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('Renders a ReservePallet with an item with a long name', () => {
    const longItemDesc = new Array(5).fill('An Item With A Very Long Item Description').join(' ');
    const mockFloorItemLongName: SectionDetailsItem = {
      ...mockLocationDetails.items.sectionItems[0], itemDesc: longItemDesc
    };
    const mockReservePalletLongFirstItemName: SectionDetailsPallet = {
      // changed due to items becoming optional on pallet
      ...mockReservePallet, items: [...(mockReservePallet.items || []), mockFloorItemLongName]
    };
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <Provider store={store}>
        <ReservePalletRow sectionId={mockSection.id} reservePallet={mockReservePalletLongFirstItemName} />
      </Provider>
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('Renders a ReservePallet with no items', () => {
    const mockReservePalletNoItems: SectionDetailsPallet = { ...mockReservePallet, items: [] };
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <Provider store={store}>
        <ReservePalletRow sectionId={mockSection.id} reservePallet={mockReservePalletNoItems} />
      </Provider>
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
