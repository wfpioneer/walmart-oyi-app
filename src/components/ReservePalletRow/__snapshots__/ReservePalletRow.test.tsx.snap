// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReservePalletRow Component Renders a ReservePallet 1`] = `
<Context.Provider
  value={
    Object {
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "@@observable": [Function],
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <ReservePalletRow
    reservePallet={
      Object {
        "items": Array [
          Object {
            "itemDesc": "Nature Valley Honey Nut Granola Bars",
            "itemNbr": 500252,
          },
          Object {
            "itemDesc": "Cabbage",
            "itemNbr": 123456,
          },
        ],
        "palletCreateTS": "2021-01-01",
        "palletId": 1,
      }
    }
  />
</Context.Provider>
`;

exports[`ReservePalletRow Component Renders a ReservePallet with an item with a long name 1`] = `
<Context.Provider
  value={
    Object {
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "@@observable": [Function],
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <ReservePalletRow
    reservePallet={
      Object {
        "items": Array [
          Object {
            "itemDesc": "Nature Valley Honey Nut Granola Bars",
            "itemNbr": 500252,
          },
          Object {
            "itemDesc": "Cabbage",
            "itemNbr": 123456,
          },
          Object {
            "itemDesc": "An Item With A Very Long Item Description An Item With A Very Long Item Description An Item With A Very Long Item Description An Item With A Very Long Item Description An Item With A Very Long Item Description",
            "itemNbr": 980078597,
            "price": 2.4,
          },
        ],
        "palletCreateTS": "2021-01-01",
        "palletId": 1,
      }
    }
  />
</Context.Provider>
`;

exports[`ReservePalletRow Component Renders a ReservePallet with no items 1`] = `
<Context.Provider
  value={
    Object {
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "@@observable": [Function],
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <ReservePalletRow
    reservePallet={
      Object {
        "items": Array [],
        "palletCreateTS": "2021-01-01",
        "palletId": 1,
      }
    }
  />
</Context.Provider>
`;
